frontend/src/components/dashboard/Analyzer.tsx

```diff
@@ -166,111 +219,166 @@ const Analyzer = ({ session }: { session: ExistingSessionExtended }) => {
+                    id="Analyzer.noGroupingAvailable"
+                    defaultMessage="Computing groups, please be patient."
+                  />
+                )}
+              </Parameters>
+            </Col>
+            <Col xs={12} lg={9}>
+              <Analysis
                 taskType={task.type}
-                groupAssignments={groupAssignments}
-                groups={groups}
+                xAxis={xAxis}
+                setXAxis={updateXAxis}
+                yAxis={yAxis}
+                setYAxis={updateYAxis}
+                categorizedDataPoints={categorizedDataPoints}
+                manualGroups={manualGroups}
+                splittingEnabled={!isAutomaticGrouping}
+                splits={splits}
+                setSplits={setSplits}
+                onSelectSolution={onSelectSolution}
               />
-            )}
-          </Col>
-        </Row>
-      )}
-    </MultiSwrContent>
+            </Col>
+            <Col xs={12}>
+              {task && (
+                <CodeComparison
+                  classId={session.klass.id}
+                  sessionId={session.id}
+                  taskId={task.id}
+                  subTaskId={selectedSubTaskId}
+                  taskType={task.type}
+                  groupAssignments={groupAssignments}
+                  groups={groups}
+                  selectedLeftGroup={selectedLeftGroup}
+                  setSelectedLeftGroup={setSelectedLeftGroup}
+                  selectedRightGroup={selectedRightGroup}
+                  setSelectedRightGroup={setSelectedRightGroup}
+                  selectedLeftSolution={selectedLeftSolution}
+                  setSelectedLeftSolution={setSelectedLeftSolution}
+                  selectedRightSolution={selectedRightSolution}
+                  setSelectedRightSolution={setSelectedRightSolution}
+                />
+              )}
+            </Col>
+          </Row>
+        )}
+      </MultiSwrContent>
+      <Modal
+        show={selectedSolutionId !== undefined}
+        onHide={() => setSelectedSolutionId(undefined)}
+        data-testid="solution-selection-modal"
+      >
+        <Modal.Header closeButton>
+          <Modal.Title>
+            {intl.formatMessage(messages.selectSolutionForComparisonTitle)}
+          </Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          {intl.formatMessage(messages.selectSolutionForComparisonDescription)}
+        </Modal.Body>
+        <Modal.Footer>
+          <Button
+            onClick={() => {
+              if (selectedSolutionId) {
+                setSelectedLeftGroup(selectedSolutionId.groupKey);
+                setSelectedLeftSolution(selectedSolutionId.solutionId);
+                setSelectedSolutionId(undefined);
+              }
+            }}
+            variant={ButtonVariant.primary}
+            data-testid="cancel-button"
+          >
+            {intl.formatMessage(messages.selectSolutionForComparisonLeft)}
```

Comment:
I'm wondering if we should also add a ↙ or ↘ symbol to make the buttons more intuitive.

Reply:
Happy to, up to them?